readme.md

```diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+    );
+    ```
+    </details>
+
 - [`NonNullable<T>`](https://github.com/Microsoft/TypeScript/blob/2961bc3fc0ea1117d4e53bc8e97fa76119bc33e3/src/lib/es5.d.ts#L1446-L1449) - Exclude `null` and `undefined` from `T`.
+    <details>
+    <summary>
+        Example
+        <a href="https://typescript-play.js.org/?target=6#code/C4TwDgpgBACg9gJ2AOQK4FsBGEFQLxQDOwCAlgHYDmUAPlORtrnQwDasDcAUFwPQBU-WAEMkUOADMowqAGNWwwoSgATCBIqlgpOOSjAAFsOBRSy1IQgr9cKJlSlW1mZYQA3HFH68u8xcoBlHA8EACEHJ08Aby4oKDBUTFZSWXjEFEYcAEIALihkXTR2YSSIAB54JDQsHAA+blj4xOTUsHSACkMzPKD3HHDHNQQAGjSkPMqMmoQASh7g-oihqBi4uNIpdraxPAI2VhmVxrX9AzMAOm2ppnwoAA4ABifuE4BfKAhWSyOTuK7CS7pao3AhXF5rV48E4ICDAVAIPT-cGQyG+XTEIgLMJLTx7CAAdygvRCA0iCHaMwarhJOIQjUBSHaACJHk8mYdeLwxtdcVAAOSsh58+lXdr7Dlcq7A3n3J4PEUdADMcspUE53OluAIUGVTx46oAKuAIAFZGQwCYAKIIBCILjUxaDHAMnla+iodjcIA">
+            Playground
+        </a>
+    </summary>
+    Works with <code>strictNullChecks</code> set to <code>true</code>. (Read more <a href="https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html">here</a>)
+
+    ```typescript
+    type PortNumber = string | number | null;
+
+    /** Part of a class definition that is used to build a server */
+    class ServerBuilder {
+        portNumber!: NonNullable<PortNumber>;
+
+        port(this: ServerBuilder, port: PortNumber): ServerBuilder {
+            if (port == null) {
+                this.portNumber = 8000;
+            } else {
+                this.portNumber = port;
+            }
+
+            return this;
+        }
+    }
+
+    const serverBuilder = new ServerBuilder();
+
+    serverBuilder
+        .port('8000')   // portNumber = '8000'
+        .port(null)     // portNumber =  8000
+        .port(3000);    // portNumber =  3000
+
+    // TypeScript Error
+    serverBuilder.portNumber = null;
+    ```
+    </details>
+
 - [`Parameters<T>`](https://github.com/Microsoft/TypeScript/blob/2961bc3fc0ea1117d4e53bc8e97fa76119bc33e3/src/lib/es5.d.ts#L1451-L1454) - Obtain the parameters of a function type in a tuple.
+    <details>
+    <summary>
+        Example
+        <a href="https://typescript-play.js.org/?target=6#code/PTAEHMBsHsCMENIBUAWBLAzqeA3ebJ5ZIBTUAMwCdoBbUAZgDoAWAKBFBQBcuAHDAFwgAJiRzFo4DIxpoAxtQzRyXRnNrAuAT14kMCtLy7B4AO1PQArqblpT4ALTbd+yoa4P6D5sADEzkgc5FBI5AGs7R3JoSgcoOEQudAxWdVMMLggYBGRk0ABeUAB3O2FoIoBuVnYAKhrQACFSims5LjRoU1AAA3JTbtAuaEGQ0DSuEgAPTOUshMhQOAArUMya4FZyVvbO0FhSpGgAcWzEAB4AMS6pidNhLAAKRmf4SikBbFMtAG0AXQBKAoAPk+WiBD1YoAopg+VwANJDQM9GK93qAAAqveA0EgTSgYS6mIGsf4fB6A-IggBKuMslFMSB0JEJIIA3ojKLT6dDGPs7g94jlUJg4UiXm8MP8qgBfarAOqgADCIXCoDQ5DGnQm0048CwvGoODQomEPTCJC03UYoHWm22HS6wVCYQAkuR0dReABRSaYLgYB5JTAfZarUXmrQfDJueykvbQaCkMygdlQzlcOldCNqrpBjAyuUKmkZ+lYbpcSiWEgDdVzHKLWArNq6ssAcnIuOCrYGBuguko2htGzSGTGKtd5AunZQPr9WEKfOEhxO8weTvCbo9fdnGQwooARB2uMF91LquuJ1PjzPfbvyRVQBwK1Wc6AaNBR7MkmRYNQihgSHxVggA">
+            Playground
+        </a>
+    </summary>
+
+    ```typescript
+    /** Bind function `fn` to the context of global object */
```

Comment:
Could you come up with a more realistic example than this? I don't think anyone actually bind to a global like this.